src/Tribe/Editor/Blocks/Rsvp.php

```diff
@@ -314,9 +317,9 @@ public function rsvp_process() {
 
 			// No point sending tickets if their current intention is not to attend
 			if ( $has_tickets && in_array( $attendee_order_status, $send_mail_stati ) ) {
```

Comment:
A bit out of scope, but both of these conditions rely on `$has_tickets` - it would be better to just test for it ahead and bail if false.
```
// Bail if we have no tickets.
if ( ! $has_tickets ) {
	continue;
}

// No point sending tickets if their current intention is not to attend
if ( in_array( $attendee_order_status, $send_mail_stati ) ) {
	tribe( 'tickets.rsvp' )->send_tickets_email( $order_id, $post_id );
} else {
	tribe( 'tickets.rsvp' )->send_non_attendance_confirmation( $order_id, $post_id );
}```

Reply:
Be sure to use `in_array()` with `true` as the third parameter in any usage

Reply:
A few questions about this suggestion.

The place where the lower part of this code block goes is inside a `if ( $send_mail )` conditional. Where are you suggesting putting the conditional with `continue`? I am only familiar with that within the context of a loop. Does `continue` just  break out of the `if()`?  Why not just change the wrapping conditional to `if ( $send_mail  && $has_tickets)` ? Or, can we `return` even earlier in the method if there is no tickets?

Reply:
Ah, I misread that code. `continue` is not appropriate there.

You could, instead, move the `$has_tickets` check up to the `$send_mail` check:
`if ( $send_mail && $has_tickets ) {...}`

Reply:
This sounds good if no further code below runs.

Reply:
we decided that since this is an ajax call, the response was need regardless.